src/test/java/org/opensearch/sdk/sample/helloworld/rest/TestHelloWorldIT.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.sample.helloworld.rest;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Before;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+public class TestHelloWorldIT extends OpenSearchRestTestCase {
+    private static final Logger logger = LogManager.getLogger(TestHelloWorldIT.class);
+
+    public static boolean initialized = false;
+    public static final String EXTENSION_INIT_URI = "/_extensions/initialize/";
+    public static final String HELLO_WORLD_EXTENSION_BASE_URI = "/_extensions/_hello-world";
+    public static final String HELLO_BASE_URI = HELLO_WORLD_EXTENSION_BASE_URI + "/hello";
+    public static final String HELLO_NAME_URI = HELLO_BASE_URI + "/%s";
+    public static final String GOODBYE_URI = HELLO_WORLD_EXTENSION_BASE_URI + "/goodbye";
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        if (!initialized) {
+            // Send initialization request
+            String helloWorldInitRequestBody = "{\"name\":\"hello-world\""
+                + ",\"uniqueId\":\"hello-world\""
+                + ",\"hostAddress\":\"127.0.0.1\""
+                + ",\"port\":\"4500\""
+                + ",\"version\":\"1.0\""
+                + ",\"opensearchVersion\":\"3.0.0\""
+                + ",\"minimumCompatibleVersion\":\"3.0.0\"}";
+            Response response = makeRequest(client(), "POST", EXTENSION_INIT_URI, null, toHttpEntity(helloWorldInitRequestBody));
+
+            assertEquals(RestStatus.ACCEPTED, restStatus(response));
+            Map<String, Object> responseMap = entityAsMap(response);
+            String initializationResponse = (String) responseMap.get("success");
+            assertEquals("A request to initialize an extension has been sent.", initializationResponse);
+            initialized = true;
+
+            // Wait for extension settings/rest handlers to complete registration before any subsequent requests
+            Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        }
```

Comment:
Instead in setUp, add it as a test since we just have one integ test right now.

Reply:
I disagree. It is necessary for the extension initialization to occur prior to the execution of any tests, therefore it should be in the `setUp` method as it is the only mechanism to enforce test execution order.

Reply:
1. We generally add common code in `setUp` which is needed for all the tests. Since we don't have any test available except the initialization there's no point of having setUp right now. For sure, it is needed later when we add more test.
2. setUp should not have a test. We should not have any assert statements in that.

Let's add a test for extension initialization, remove setUp(add a TODO to move the initialization to setUp once we have more integ test)

Reply:
sure sounds good, ill go ahead and make this change